src/services/goals.js

```diff
@@ -1686,13 +1686,35 @@ export async function removeRemovedRecipientsGoals(removedRecipientIds, report)
   }
 
   if (Array.isArray(objectivesToDelete) && objectivesToDelete.length > 0) {
-    await Objective.destroy({
+    const objectivesToDefinitelyDestroy = await Objective.findAll({
+      attributes: [
+        'id',
+        'onApprovedAR',
+      ],
       where: {
         id: objectivesToDelete,
         onApprovedAR: false,
       },
-      individualHooks: true,
     });
+
+    if (Array.isArray(objectivesToDefinitelyDestroy) && objectivesToDefinitelyDestroy.length > 0) {
+      const objectiveIdsToDestroy = objectivesToDefinitelyDestroy.map((o) => o.id);
+
+      await ObjectiveFile.destroy({
```

Comment:
Maybe a dumb question but going forward we will need to make sure to add any other objective tables here?

Reply:
I don't think that's a dumb question, test verifies we don't need to add topics here but I'll go ahead and add resources so we don't have to worry about it